From ac60faa895b28e7a4e58591781315b1aa26bc3e9 Mon Sep 17 00:00:00 2001
From: jinghua <jinghua@marvell.com>
Date: Mon, 28 Mar 2016 21:19:49 +0800
Subject: [PATCH 0334/2241] fix: sdio: a3700: add gpio regulator for a3700 db
 sdio fdt node

- add GPIO regulator to switch SDIO voltage for compatibility.
- some SD cards need 1.8V while others need 3.3V, depends on
  negotiation between SDIO controller and SD card.

Change-Id: Ib770ff502337c48359c3151eb2e33c661dc5430e
Signed-off-by: jinghua <jinghua@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/28571
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Omri Itach <omrii@marvell.com>
Reviewed-by: Nadav Haklai <nadavh@marvell.com>
---
 arch/arm64/boot/dts/marvell/armada-3720-db.dts | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/arch/arm64/boot/dts/marvell/armada-3720-db.dts b/arch/arm64/boot/dts/marvell/armada-3720-db.dts
index bd59c6f..ada307b 100644
--- a/arch/arm64/boot/dts/marvell/armada-3720-db.dts
+++ b/arch/arm64/boot/dts/marvell/armada-3720-db.dts
@@ -127,6 +127,7 @@
 				broken-cd;
 				wp-inverted;
 				bus-width = <4>;
+				vqmmc-supply = <&vcc_sd_reg1>;
 				status = "okay";
 
 				pinctrl-names = "default";
@@ -148,6 +149,20 @@
 			sata@e0000 {
 			       status = "okay";
 			};
+			vcc_sd_reg1: regulator@0 {
+				compatible = "regulator-gpio";
+				regulator-name = "vcc_sd1";
+				regulator-min-microvolt = <1800000>;
+				regulator-max-microvolt = <3300000>;
+				regulator-boot-on;
+
+				gpios = <&gpio_sb 22 GPIO_ACTIVE_HIGH>;
+				gpios-states = <0>;
+				states = <1800000 0x1
+					  3300000 0x0>;
+
+				enable-active-high;
+			};
 		};
 	};
 };
-- 
2.7.4

